bottlecap/src/lifecycle/flush_control.rs

```diff
@@ -35,25 +35,17 @@ impl FlushControl {
     pub fn should_flush_end(&mut self) -> bool {
         // previously: would return true if flush_strategy is not Periodically
         // !matches!(self.flush_strategy, FlushStrategy::Periodically(_))
-        let now = match time::SystemTime::now().duration_since(time::UNIX_EPOCH) {
-            Ok(now) => now.as_secs(),
-            Err(e) => {
-                debug!("Failed to get current time: {:?}", e);
-                return false;
-            }
-        };
+        let now = time::SystemTime::now()
+            .duration_since(time::UNIX_EPOCH)
+            .expect("unable to poll clock, unrecoverable")
+            .as_secs();
         self.invocation_times.add(now);
         match &self.flush_strategy {
             FlushStrategy::End | FlushStrategy::EndPeriodically(_) => true,
             FlushStrategy::Periodically(_) => false,
             FlushStrategy::Default => {
                 if self.invocation_times.should_adapt_to_periodic(now) {
-                    let should_periodic_flush = self.should_periodic_flush(now, TWENTY_SECONDS);
-                    debug!(
-                        "Adapting over to periodic flush strategy. should_periodic_flush: {}",
-                        should_periodic_flush
-                    );
-                    return should_periodic_flush;
+                    return false;
```

Comment:
Not sure I understand this

Reply:
nope, accident

Reply:
previously this method would control whether or not to switch over to flushing periodically _as well as_ determining if we've switched to periodic and it's time to flush. That's partially why the main loop was so confusing.

Instead, we've lifted that logic to the main loop, so this method can now be simplified. It only determines if the user specified they want to flush at end, or if we're on the default strategy – if we've seen enough invocations to flip over to periodic flushing. If the answer is yes, then we should not flush at end.